tests/tools.vitruv.applications.pcmumlclass.tests/src/tools/vitruv/applications/pcmumlclass/tests/MediaStoreRepositoryCreationTest.xtend

```diff
@@ -165,4 +175,331 @@ class MediaStoreRepositoryCreationTest extends LegacyPcmUmlClassApplicationTest
+				it.name == originalComponentImpl.name
+			]
+			assertNotNull(generatedComponentImpl)
+
+			// merge everything except operations which are separately handled, because some of the constructor parameters will be generated.
+			resolveElements(originalComponentImpl, generatedModel.eResource, "ownedOperation", "interfaceRealization")
+			mergeElements(originalComponentImpl, generatedComponentImpl, "ownedOperation", "interfaceRealization")
+			for (originalRealization : originalComponentImpl.interfaceRealizations.clone) {
+				resolveElements(originalRealization, generatedModel.eResource)
+				originalComponentImpl.interfaceRealizations -= originalRealization
+				originalRealization.clients -= originalComponentImpl // needs to be manually cleared, or else originalComponentImpl is still set as a client and causes UUID error
+				generatedComponentImpl.interfaceRealizations += originalRealization
+			}
+		]
+
+		// handle errors that are produced if the changeUmlView doesn't produce an actual change
+		try {
+			changeUmlView [
+				var generatedModel = UmlQueryUtil.claimUmlModel(it, modelName)
+				var generatedRepositoryPackage = generatedModel.nestedPackages.head
+				assertNotNull(generatedRepositoryPackage)
+				var generatedComponentPackage = generatedRepositoryPackage.nestedPackages.findFirst [
+					it.name == originalComponentPackage.name
+				]
+				assertNotNull(generatedComponentPackage)
+				var generatedComponentImpl = generatedComponentPackage.packagedElements.filter(Class).findFirst [
+					it.name == originalComponentImpl.name
+				]
+				assertNotNull(generatedComponentImpl)
+
+				// merge the original with the generated constructor
+				val originalConstructor = originalComponentImpl.ownedOperations.findFirst [
+					it.name == originalComponentImpl.name
+				]
+				val generatedConstructor = generatedComponentImpl.ownedOperations.findFirst [
+					it.name == originalComponentImpl.name
+				]
+				assertNotNull(originalConstructor)
+				assertNotNull(generatedConstructor)
+				resolveElements(originalConstructor, generatedModel.eResource, "ownedParameter")
+				mergeElements(originalConstructor, generatedConstructor, "ownedParameter")
+				for (originalParameter : originalConstructor.ownedParameters.clone) {
+					val generatedParameter = generatedConstructor.ownedParameters.findFirst [
+						it.name == originalParameter.name
+					]
+					resolveElements(originalParameter, generatedModel.eResource, "name")
+					if (generatedParameter !== null) {
+						mergeElements(originalParameter, generatedParameter, "name")
+					} else {
+						generatedConstructor.ownedParameters += originalParameter
+					}
+				}
+				// move/copy the remaining operations
+				for (originalOperation : originalComponentImpl.ownedOperations.clone.filter [
+					it.name != originalComponentImpl.name
+				]) {
+					generatedComponentImpl.ownedOperations += originalOperation
+				}
+			]
+		} catch (IllegalArgumentException e) {
```

Comment:
That's patching a symptom. Instead, empty changes (which do not contain concrete changes) should not throw exceptions (they also occur with ChangeDerivingViews and multiple metamodels, which do not necessarily create changes in each other model in every change propagation phase). This should be recorded in a new issue.